packages/ember-glimmer/tests/integration/input-test.js

```diff
@@ -0,0 +1,155 @@
+  }
+
+  runPropertyTest(propertyName, values) {
+    let attributeName = propertyName;
+    let template = `<input ${attributeName}={{value}}>`;
+    this.render(template, { value: values[0] });
+    this.assertPropertyHasValue(propertyName, values[0], `${propertyName} is set on initial render`);
+
+    this.runTask(() => this.rerender());
+    this.assertPropertyHasValue(propertyName, values[0], `${propertyName} is set on noop rerender`);
+
+    this.setComponentValue(values[1]);
+    this.assertPropertyHasValue(propertyName, values[1], `${propertyName} is set on rerender`);
+
+    this.setComponentValue(values[0]);
+    this.assertPropertyHasValue(propertyName, values[0], `${propertyName} can be set back to the initial value`);
+  }
+
+  ['@test input disabled attribute']() {
+    this.runPropertyTest('disabled', [false, true]);
+  }
+
+  ['@test input value attribute']() {
+    this.runPropertyTest('value', ['foo', 'bar']);
+  }
+
+  ['@htmlbars input placeholder attribute']() {
+    this.runAttributeTest('placeholder', ['foo', 'bar']);
+  }
+
+  ['@test input name attribute']() {
+    this.runAttributeTest('name', ['nam', 'name']);
+  }
+
+  ['@htmlbars input maxlength attribute']() {
+    this.runAttributeTest('maxlength', [2, 3]);
+  }
+
+  ['@test input size attribute']() {
+    this.runAttributeTest('size', [2, 3]);
+  }
+
+  ['@htmlbars input tabindex attribute']() {
+    this.runAttributeTest('tabindex', [2, 3]);
+  }
+
+  ['@htmlbars cursor position is not lost when updating content']() {
+    let template = `<input value={{value}}>`;
+    this.render(template, { value: 'hola' });
+
+    this.setDOMValue('hello');
+    this.setSelectionRange(1, 3);
+
+    this.setComponentValue('hello');
+
+    this.assertSelectionRange(1, 3);
+
+    // Note: We should eventually get around to testing reseting, however
+    // browsers handle `selectionStart` and `selectionEnd` differently
+    // when are synthetically testing movement of the cursor.
+  }
+
+  ['@test input can be updated multiple times']() {
+    let template = `<input value={{value}}>`;
+    this.render(template, { value: 'hola' });
+
+    this.assertValue('hola', 'Value is initialised');
+
+    this.setComponentValue('');
+    this.assertValue('', 'Value is set in the DOM');
+
+    this.setDOMValue('hola');
+    this.setComponentValue('hola');
+    this.assertValue('hola', 'Value is updated the first time');
+
+    this.setComponentValue('');
+    this.assertValue('', 'Value is updated the second time');
+  }
+
+  ['@test DOM is SSOT if value is set']() {
+    let template = `<input value={{value}}>`;
+    this.render(template, { value: 'hola' });
+
+    this.assertValue('hola', 'Value is initialised');
+
+    this.setComponentValue('hello');
+
+    this.assertValue('hello', 'Value is initialised');
+
+    this.setDOMValue('hola');
+
+    this.assertValue('hola', 'DOM is used');
+
+    this.setComponentValue('hello');
+
+    this.assertValue('hola', 'DOM is used');
```

Comment:
This is not correct. It should be `hello`. While it is true that the DOM is the SSOT, this doesn't mean that mutation via `setComponentValue` cannot mutate the SSOT.